hyperspace/core/README.md

```diff
@@ -17,15 +17,48 @@ alongside optional metric handlers and starts the relayer loop.
 
 The relayer loops awaits finality events from the finality subscription of the chain handlers.  
 Whenever a finality event is received, the latest ibc events are queried using `query_latest_ibc_events`.  
-These events are then parsed into the appropriate outgoing IBC messages, and sent off to the counterparty chain. 
-
-### Packet Timeouts
+These events are then parsed into the appropriate outgoing IBC messages, and sent off to the counterparty chain.
 
 The `query_ready_and_timed_out_packets` which queries a chain and  
 produces all packet messages that have passed the connection delay check.
 It also returns timed out packet messages that have passed the connection delay check.  
 
-// todo: let's talk more about how we handle timeouts
+### Connection delay
+ 
+The relayer needs to submit packets with a proof fetched at a height where the equivalent client consensus state on the  
```

Comment:
Can we reword this long sentence? I'm not sure I understand it - perhaps it's too long?

Reply:
I'll use pseudo code to explain it instead.

Reply:
thanks!